blog post terraform partnership (#508)
kestra-io/docs: content/blogs/2023-09-18-kestra-terraform-partnership.md

---

> title: "A Game-Changer in Everything-as-Code: Kestra and Terraform by HashiCorp Join Forces"

Reviewer: ```suggestion
title: "Kestra and Terraform by HashiCorp Join Forces to Enable Everything-as-Code"
```

Author: that's better!

> description: "Discover the technology partnership between Kestra and Terraform by HashiCorp. Experience simplified operational workflows across data and infrastructure, seamless CI/CD integrations, and a Terraform provider with already 200,000 downloads."

Reviewer: ```suggestion
description: "Introducing the technological partnership with Terraform by Hashicorp — Kestra is now a verified Terraform partner. Learn in this post how Kestra's battle-tested Terraform provider, with over 200,000 downloads, simplifies CI/CD for operational and analytical workflows."
```

date: 2023-09-18T12:00:00
category: Company News
author:
  name: Martin-Pierre Roset
  image: "mproset"
image: /blogs/2023-09-18-kestra-terraform-partnership.jpg
---

> We're thrilled to reveal that Kestra has formed a technology partnership with [Terraform](https://www.terraform.io/) by HashiCorp! We are both united by a shared vision to simplify data and infrastructure complexities. And to add to this amazing news, our Terraform provider is already a hit, with **200,000 downloads** making it one of the top Terraform providers! 

Reviewer: ```suggestion
We're thrilled to announce a technology partnership with [Terraform](https://www.terraform.io/) by HashiCorp. We share a common vision of simplifying deployment patterns by relying on Infrastructure as Code. We are excited to further extend the Everything-as-Code paradigm to your mission-critical operational and analytical workflows.
```

> >“Integrating Kestra with Terraform enables a cohesive approach to automating your workflows. It's not just about tooling; it's about  streamlining operations in a way that resonates with core DevOps principles.”

Reviewer: ```suggestion
>“The Terraform provider allows our users to express not only software and workflows but also their underlying infrastructure configuration in code based on proven DevOps principles.”
```

> >“Integrating Kestra with Terraform enables a cohesive approach to automating your workflows. It's not just about tooling; it's about  streamlining operations in a way that resonates with core DevOps principles.”

Reviewer: ```suggestion
>“Integrating Kestra with Terraform enables a cohesive approach to automating your workflows. It's not just about tooling; it's about  streamlining operations in a way that is aligned with core DevOps principles.”
```

> *Ludovic Dehon CTO @ Kestra*

Reviewer: ```suggestion
*Ludovic Dehon — the CTO at Kestra*
```

---

> ## Why Terraform and Kestra? A Technologic Alignment ##

Reviewer: ```suggestion
## Why Terraform 
```

> Our partnership with Terraform goes beyond just working together; it's about synchronizing our strengths. Both of our technologies aim to make your operational tasks less complicated. With this partnership, you can easily configure Kestra workflows and tasks within your Terraform settings. This coordination makes scaling straightforward and keeps your operational changes well-documented through a code-based approach.

Reviewer: ```suggestion
Our partnership with Terraform extends beyond technical integration. We both share a common vision of expressing not only software but also its infrastructure and configuration in code. With this partnership, you can manage even Zapier-style workflows according to proven software engineering practices.
```

--- 

> ## Elevating Your Data Workflow With Kestra's Terraform Provider ##

Reviewer: ```suggestion
## Elevating Your Workflows With Kestra's Terraform Provider 
```

![Terraform Kestra](/blogs/2023-09-18-kestra-terraform-partnership/terraform-kestra.png)

> Our Terraform provider is a significant step in aligning data orchestration with infrastructure-as-code best practices. With this integration, configuring Kestra workflows and tasks directly within your Terraform scripts is a seamless experience. It's an integrated approach to streamline and automate all your flows.

Reviewer: ```suggestion
Our Terraform provider is a significant step in aligning workflow orchestration with infrastructure-as-code best practices. With this integration, you can seamlessly manage your entire Kestra environment in code, including flows, triggers, tasks, input variables, secrets, users, and namespaces. Software engineers manage their entire Kestra infrastructure along with their other applications with Terraform to ensure a robust software delivery process.
```

> ### Breaking the Production Barrier: The Kestra and Terraform Solution ###

Reviewer: ```suggestion
### Breaking the Production Barrier: The Kestra and Terraform Solution
```

Moving data pipelines from development to production comes with its own set of challenges. Be it environment setups, database schema access, or cloud resource permissions, the list seems endless. The alliance between Kestra and Terraform aims to bridge this gap by integrating open-source tools that make this transition easier for developers. Learn more about [Everything as Code for Data Infrastructure solution](https://levelup.gitconnected.com/programmable-data-infrastructure-is-finally-within-reach-83fb8e6392ac).

> ### Supercharge Your CI/CD with Terraform & Kestra ###

Reviewer: ```suggestion
### Supercharge Your CI/CD with Terraform & Kestra
```

> Bring Kestra into your CI/CD pipelines using tools like GitHub Actions. By doing so you can automate your infrastructure provisioning but also deploy your Kestra flows in the CI/CD model. With a GitHub Actions workflow, you can also validate and deploy flows directly into different namespaces ensuring that your data workflows are as agile and adaptable as your infrastructure.

Reviewer: ```suggestion
Bring Kestra into your CI/CD pipelines using Terraform Cloud or GitHub Actions. By doing so, you can automate your infrastructure provisioning and maintenance. With a GitHub Actions workflow, you can also validate and deploy flows directly into specific namespaces, ensuring that your mission-critical workflows are as agile and adaptable as your infrastructure.
```

<br>
<iframe width="560" height="315" src="https://www.youtube.com/embed/videoseries?si=jXSj49xwStR5Wqk8&amp;list=PLEK3H8YwZn1olYeQ7qbGdY5XQySKCNLUr" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>
<br>

> ### Unlocking Data Efficiency: 5 Million Executions and 4,000+ Flows with Kestra and Terraform ###

Reviewer: ```suggestion
### 10x Your Workflow Automation and Software Delivery
```

those numbers are not encouraging - executions are not deployed from terraform, and 4k flows seems like not much, I'd cut it

**Leroy Merlin France** achieved new heights in data agility by integrating Kestra and Terraform, both employing declarative language and fully embracing Git-based version control. They used Terraform for automated deployments of BigQuery assets like datasets, tables, views, and ACLs, all part of a streamlined CI/CD process. Upon adopting Kestra, a custom Terraform provider was developed to bring the same level of version control and automation to Kestra’s resources. This harmonized architecture has led to over **5 million pipeline executions** each month, and more than **4.000+ data flows created**, and a staggering 900% increase in data activity, all while fostering team-wide confidence during production deployments. [Learn more about this amazing use case](https://kestra.io/blogs/2023-08-16-datamesh)

--- 

> ## What’s Next ##

Reviewer: ```suggestion
## What’s Next 
```

> Our collaboration is just the beginning. Your insights and contributions can greatly influence the utility and success of this integration. Stay tuned for more updates and opportunities.

Reviewer: ```suggestion
Our collaboration is just the beginning. Your insights and contributions can greatly influence the utility and success of this integration. Share your feedback via [Kestra Community Slack](https://kestra.io/slack) or open [a GitHub issue](https://github.com/kestra-io/kestra).
```

it's weird to mention that user's insights/feedback is welcome without showing how they can provide it

Author: Yep right! just dumb copy pasted the final message with the link

> Hungry for more? Dive into Kestra's Terraform Provider [Documentation](https://kestra.io/docs/terraform).

Reviewer: ```suggestion
If you want to learn more, dive into Kestra's Terraform Provider [Documentation](https://kestra.io/docs/terraform).
```

not a fan of this language "hungry for more?" :P up to you though

Author: Don't bring anything here since it's link dropping agree

> If you have any questions, reach out via [Kestra Community Slack](https://kestra.io/slack) or open [a GitHub issue](https://github.com/kestra-io/kestra).

Reviewer: ```suggestion
```

mentioned above
